File: FCSeafood/FCSeafood.BusinessObjects/Models/Events/UserModel.cs
namespace FCSeafood.BusinessObjects.Models.Events;

public class UserModel {
    public Guid Id { get; set; }
    public string FirstName { get; set; } = "";
    public string LastName { get; set; } = "";
    public string Email { get; set; } = "";
    public RoleTModel Role { get; set; } = new();
    public GenderTModel Gender { get; set; } = new();
    public bool IsActive { get; set; } = true;
    public bool IsVerified { get; set; }
    public string? Phone { get; set; }
    public string? ImagePath { get; set; }
    public DateTime? DateOfBirth { get; set; }
    public AddressModel? Address { get; set; }
    public DateTime CreatedDate { get; set; } = DateTime.Now;
    public DateTime UpdatedDate { get; set; } = DateTime.Now;

    public string GetFullName => $"{FirstName} {LastName}";
}

File: FCSeafood-Angular/src/app/common/models/user.model.ts
import { RoleTModel } from "@common-models/role-type.model";
import { GenderTModel } from "@common-models/gender-type.model";
import { AddressModel } from "@common-models/address.model";

export class UserModel {
  id: string = ''; // guid
  firstName: string = '';
  lastName: string = '';
  email: string = '';
  role: RoleTModel = new RoleTModel();
  gender: GenderTModel = new GenderTModel();
  isActive: boolean = false;
  isVerified: boolean = false;
  phone: string | null = null;
  imagePath: string | null = null;
  dateOfBirth: Date | null = null;
  address: AddressModel | null = null;
  createdDate: Date = new Date();
  updatedDate: Date = new Date();
}
